gridstatus/eia.py

```diff
@@ -741,14 +755,49 @@ def _handle_fuel_type_data(df):
         values="MW",
     ).reset_index()
 
+    # Handle case-insensitive column duplicates (e.g. "Pumped Storage","Pumped storage")
+    df.columns = df.columns.str.lower()
+    for col in df.columns[df.columns.duplicated()]:
+        total = df[col].sum(axis="columns")
+        df = df.drop(columns=col)
+        df[col] = total
+
+    df.columns = df.columns.str.title()
+
+    possible_fuel_mix_columns = [
+        "Battery",
+        "Battery Storage",
+        "Coal",
+        "Hydro",
+        "Natural Gas",
+        "Nuclear",
+        "Other",
+        "Petroleum",
+        "Pumped Storage",
+        "Solar",
+        "Solar Battery",
+        "Solar With Integrated Battery Storage",
+        "Unknown Energy",
+        "Unknown Energy Storage",
+        "Wind",
+    ]
+
+    for col in possible_fuel_mix_columns:
+        if col not in df.columns:
+            # This has to be np.nan not pd.NA because we are converting to float
+            df[col] = np.nan
+
     fuel_mix_cols = df.columns[4:]
 
-    # nans after pivot because not
-    # all respondents have all fuel types
+    # nans after pivot because not all respondents have all fuel types
+    # Why are we filling these with 0? A value of 0 means something different than NaN
```

Comment:
@kmax12 Why are the na values being filled with 0? I would think they should be kept as na to avoid creating data values.

Reply:
i think my logic was that if the fuel didn't get reported, it means it wasn't generated, hence it is okay to say zero. 

I'm okay if you want to change it to null.

if you do, we'll want to put in the description of the dataset that null means that it wasn't reported.